itk tests don't compile with visual studio

I tried the superbuild with visual studio 2012 (64bit). After modifiying a few things for a succesful installation of boost, the examples compile but I still have problems with the projects 
ITKHeaderTest1
ITKHeaderTest2
ITKHeaderTest3
ITKHeaderTest4

Error   6   error C2874: using-declaration causes a multiple declaration of 'boost::reference_wrapper'  E:\dev\statismo0.10\superbuild\INSTALL\include\boost-1_55\boost\tr1\functional.hpp  31  ITKHeaderTest1
Error   10  error C2953: 'std::result_of' : class template has already been defined E:\dev\statismo0.10\superbuild\INSTALL\include\boost-1_55\boost\tr1\functional.hpp  54  ITKHeaderTest1
Error   13  error C2874: using-declaration causes a multiple declaration of 'boost::is_bind_expression' E:\dev\statismo0.10\superbuild\INSTALL\include\boost-1_55\boost\tr1\functional.hpp  79  ITKHeaderTest1
Error   14  error C2874: using-declaration causes a multiple declaration of 'boost::is_placeholder' E:\dev\statismo0.10\superbuild\INSTALL\include\boost-1_55\boost\tr1\functional.hpp  80  ITKHeaderTest1
Error   17  error C2874: using-declaration causes a multiple declaration of 'boost::bad_function_call'  E:\dev\statismo0.10\superbuild\INSTALL\include\boost-1_55\boost\tr1\functional.hpp  110 ITKHeaderTest1

--- a/modules/ITK/include/itkConditionalModelBuilder.h
+++ b/modules/ITK/include/itkConditionalModelBuilder.h
@@ -80,7 +80,7 @@ class ConditionalModelBuilder : public Object {
     }
 
     template <class F>
-    typename std::tr1::result_of<F()>::type callstatismoImpl(F f) const {
+    typename boost::result_of<F()>::type callstatismoImpl(F f) const {
         try {
             return f();
         } catch (statismo::StatisticalModelException& s) {
@@ -96,7 +96,7 @@ class ConditionalModelBuilder : public Object {
                   float noiseVariance,
                   double modelVarianceRetained
                  ) {
-        statismo::StatisticalModel<Representer>* model_statismo = callstatismoImpl(std::tr1::bind(&ImplType::BuildNewModel, this->m_impl, SampleDataStructureList, surrogateTypes, conditioningInfo, noiseVariance, modelVarianceRetained));
+        statismo::StatisticalModel<Representer>* model_statismo = callstatismoImpl(boost::bind(&ImplType::BuildNewModel, this->m_impl, SampleDataStructureList, surrogateTypes, conditioningInfo, noiseVariance, modelVarianceRetained));
         typename StatisticalModel<Representer>::Pointer model_itk = StatisticalModel<Representer>::New();
         model_itk->SetstatismoImplObj(model_statismo);
         return model_itk;

--- a/modules/ITK/include/itkDataManager.h
+++ b/modules/ITK/include/itkDataManager.h
@@ -39,7 +39,8 @@
 #ifndef ITK_DATAMANAGER_H_
 #define ITK_DATAMANAGER_H_
 
-#include <boost/tr1/functional.hpp>
+#include <boost/bind.hpp>
+#include <boost/utility/result_of.hpp>
 
 #include "itkObject.h"
 #include "itkObjectFactory.h"
@@ -73,7 +74,7 @@ class DataManager : public Object {
     typedef statismo::Representer<T> RepresenterType;
 
     template <class F>
-    typename std::tr1::result_of<F()>::type callstatismoImpl(F f) const {
+    typename boost::result_of<F()>::type callstatismoImpl(F f) const {
         if (m_impl == 0) {
             itkExceptionMacro(<< "Model not properly initialized. Maybe you forgot to call SetRepresenter");
         }
@@ -111,7 +112,7 @@ class DataManager : public Object {
     }
 
     void AddDataset(typename RepresenterType::DatasetType* ds, const char* filename) {
-        callstatismoImpl(std::tr1::bind(&ImplType::AddDataset, this->m_impl, ds, filename));
+        callstatismoImpl(boost::bind(&ImplType::AddDataset, this->m_impl, ds, filename));
     }
 
     void Load(const char* filename) {
@@ -123,11 +124,11 @@ class DataManager : public Object {
     }
 
     void Save(const char* filename) {
-        callstatismoImpl(std::tr1::bind(&ImplType::Save, this->m_impl, filename));
+        callstatismoImpl(boost::bind(&ImplType::Save, this->m_impl, filename));
     }
 
     typename statismo::DataManager<T>::DataItemListType GetData() const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::GetData, this->m_impl));
+        return callstatismoImpl(boost::bind(&ImplType::GetData, this->m_impl));
     }
 
 

--- a/modules/ITK/include/itkDataManagerWithSurrogates.h
+++ b/modules/ITK/include/itkDataManagerWithSurrogates.h
@@ -39,7 +39,7 @@
 #ifndef ITK_DATAMANAGER_WITH_SURROGATES_H_
 #define ITK_DATAMANAGER_WITH_SURROGATES_H_
 
-#include <boost/tr1/functional.hpp>
+#include <boost/utility/result_of.hpp>
 
 #include "itkObject.h"
 #include "itkObjectFactory.h"
@@ -70,7 +70,7 @@ class DataManagerWithSurrogates : public statismo::DataManager<Representer> {
     typedef statismo::DataManagerWithSurrogates<Representer> ImplType;
 
     template <class F>
-    typename std::tr1::result_of<F()>::type callstatismoImpl(F f) const {
+    typename boost::result_of<F()>::type callstatismoImpl(F f) const {
         if (m_impl == 0) {
             itkExceptionMacro(<< "Model not properly initialized. Maybe you forgot to call SetParameters");
         }
@@ -112,7 +112,7 @@ class DataManagerWithSurrogates : public statismo::DataManager<Representer> {
     void AddDatasetWithSurrogates(typename Representer::DatasetConstPointerType ds,
                                   const char* datasetURI,
                                   const char* surrogateFilename) {
-        callstatismoImpl(std::tr1::bind(&ImplType::AddDatasetWithSurrogates, this->m_impl, ds, datasetURI, surrogateFilename));
+        callstatismoImpl(boost::bind(&ImplType::AddDatasetWithSurrogates, this->m_impl, ds, datasetURI, surrogateFilename));
     }
 
 

--- a/modules/ITK/include/itkPCAModelBuilder.h
+++ b/modules/ITK/include/itkPCAModelBuilder.h
@@ -80,7 +80,7 @@ class PCAModelBuilder : public Object {
     }
 
     template <class F>
-    typename std::tr1::result_of<F()>::type callstatismoImpl(F f) const {
+    typename boost::result_of<F()>::type callstatismoImpl(F f) const {
         try {
             return f();
         } catch (statismo::StatisticalModelException& s) {
@@ -91,7 +91,7 @@ class PCAModelBuilder : public Object {
 
 
     typename StatisticalModel<Representer>::Pointer BuildNewModel(DataItemListType DataItemList, float noiseVariance, bool computeScores = true) {
-        statismo::StatisticalModel<Representer>* model_statismo = callstatismoImpl(std::tr1::bind(&ImplType::BuildNewModel, this->m_impl, DataItemList, noiseVariance, computeScores));
+        statismo::StatisticalModel<Representer>* model_statismo = callstatismoImpl(boost::bind(&ImplType::BuildNewModel, this->m_impl, DataItemList, noiseVariance, computeScores));
         typename StatisticalModel<Representer>::Pointer model_itk = StatisticalModel<Representer>::New();
         model_itk->SetstatismoImplObj(model_statismo);
         return model_itk;

--- a/modules/ITK/include/itkPosteriorModelBuilder.h
+++ b/modules/ITK/include/itkPosteriorModelBuilder.h
@@ -71,7 +71,7 @@ class PosteriorModelBuilder : public Object {
 
 
     template <class F>
-    typename std::tr1::result_of<F()>::type callstatismoImpl(F f) const {
+    typename boost::result_of<F()>::type callstatismoImpl(F f) const {
         try {
             return f();
         } catch (statismo::StatisticalModelException& s) {
@@ -102,7 +102,7 @@ class PosteriorModelBuilder : public Object {
 
     typename StatisticalModelType::Pointer BuildNewModelFromModel(const StatisticalModelType* model, const PointValueListType& pointValues, double pointValuesNoiseVariance,  bool computeScores=true) {
         StatismoStatisticalModelType* model_statismo = model->GetstatismoImplObj();
-        StatismoStatisticalModelType* new_model_statismo = callstatismoImpl(std::tr1::bind(
+        StatismoStatisticalModelType* new_model_statismo = callstatismoImpl(boost::bind(
                     static_cast<StatismoStatisticalModelType* (ImplType::*) (const StatismoStatisticalModelType* model, const PointValueListType& pointValues, double pointValuesNoiseVariance,  bool computeScores) const > (&ImplType::BuildNewModelFromModel),
                     this->m_impl, model_statismo, pointValues, pointValuesNoiseVariance, computeScores));
         typename StatisticalModelType::Pointer model_itk = StatisticalModelType::New();
@@ -111,7 +111,7 @@ class PosteriorModelBuilder : public Object {
     }
 
     typename StatisticalModelType::Pointer BuildNewModel(DataItemListType DataItemList, const PointValueListType& pointValues, double pointValuesNoiseVariance, double noiseVariance) {
-        StatismoStatisticalModelType* model_statismo = callstatismoImpl(std::tr1::bind(&ImplType::BuildNewModel, this->m_impl, DataItemList ,pointValues, pointValuesNoiseVariance, noiseVariance));
+        StatismoStatisticalModelType* model_statismo = callstatismoImpl(boost::bind(&ImplType::BuildNewModel, this->m_impl, DataItemList ,pointValues, pointValuesNoiseVariance, noiseVariance));
         typename StatisticalModelType::Pointer model_itk = StatisticalModelType::New();
         model_itk->SetstatismoImplObj(model_statismo);
         return model_itk;
@@ -119,7 +119,7 @@ class PosteriorModelBuilder : public Object {
 
     typename StatisticalModelType::Pointer BuildNewModelFromModel(const StatisticalModelType* model, const PointValueWithCovarianceListType& pointValuesWithCovariance,  bool computeScores=true) {
         StatismoStatisticalModelType* model_statismo = model->GetstatismoImplObj();
-        StatismoStatisticalModelType* new_model_statismo = callstatismoImpl(std::tr1::bind(
+        StatismoStatisticalModelType* new_model_statismo = callstatismoImpl(boost::bind(
                     static_cast<StatismoStatisticalModelType* (ImplType::*) (const StatismoStatisticalModelType* model, const PointValueWithCovarianceListType& pointValuesWithCovariance,  bool computeScores) const > (&ImplType::BuildNewModelFromModel),
                     this->m_impl, model_statismo, pointValuesWithCovariance, computeScores));
         typename StatisticalModelType::Pointer model_itk = StatisticalModelType::New();
@@ -128,7 +128,7 @@ class PosteriorModelBuilder : public Object {
     }
 
     typename StatisticalModelType::Pointer BuildNewModel(const DataItemListType& DataItemList, const PointValueWithCovarianceListType& pointValuesWithCovariance, 	double noiseVariance) {
-        StatismoStatisticalModelType* model_statismo = callstatismoImpl(std::tr1::bind(&ImplType::BuildNewModel, this->m_impl, DataItemList, pointValuesWithCovariance, noiseVariance));
+        StatismoStatisticalModelType* model_statismo = callstatismoImpl(boost::bind(&ImplType::BuildNewModel, this->m_impl, DataItemList, pointValuesWithCovariance, noiseVariance));
         typename StatisticalModelType::Pointer model_itk = StatisticalModelType::New();
         model_itk->SetstatismoImplObj(model_statismo);
         return model_itk;

--- a/modules/ITK/include/itkReducedVarianceModelBuilder.h
+++ b/modules/ITK/include/itkReducedVarianceModelBuilder.h
@@ -68,7 +68,7 @@ class ReducedVarianceModelBuilder : public Object {
 
 
     template <class F>
-    typename std::tr1::result_of<F()>::type callstatismoImpl(F f) const {
+    typename boost::result_of<F()>::type callstatismoImpl(F f) const {
         try {
             return f();
         } catch (statismo::StatisticalModelException& s) {
@@ -90,7 +90,7 @@ class ReducedVarianceModelBuilder : public Object {
 
     typename StatisticalModel<Representer>::Pointer BuildNewModelWithLeadingComponents(const StatisticalModel<Representer>* model, unsigned numberOfPrincipalComponents) {
         statismo::StatisticalModel<Representer>* model_statismo = model->GetstatismoImplObj();
-        statismo::StatisticalModel<Representer>* new_model_statismo = callstatismoImpl(std::tr1::bind(&ImplType::BuildNewModelWithLeadingComponents, this->m_impl, model_statismo, numberOfPrincipalComponents));
+        statismo::StatisticalModel<Representer>* new_model_statismo = callstatismoImpl(boost::bind(&ImplType::BuildNewModelWithLeadingComponents, this->m_impl, model_statismo, numberOfPrincipalComponents));
         typename StatisticalModel<Representer>::Pointer model_itk = StatisticalModel<Representer>::New();
         model_itk->SetstatismoImplObj(new_model_statismo);
         return model_itk;
@@ -98,7 +98,7 @@ class ReducedVarianceModelBuilder : public Object {
 
     typename StatisticalModel<Representer>::Pointer BuildNewModelWithVariance(const StatisticalModel<Representer>* model, double totalVariance) {
         statismo::StatisticalModel<Representer>* model_statismo = model->GetstatismoImplObj();
-        statismo::StatisticalModel<Representer>* new_model_statismo = callstatismoImpl(std::tr1::bind(&ImplType::BuildNewModelWithVariance, this->m_impl, model_statismo, totalVariance));
+        statismo::StatisticalModel<Representer>* new_model_statismo = callstatismoImpl(boost::bind(&ImplType::BuildNewModelWithVariance, this->m_impl, model_statismo, totalVariance));
         typename StatisticalModel<Representer>::Pointer model_itk = StatisticalModel<Representer>::New();
         model_itk->SetstatismoImplObj(new_model_statismo);
         return model_itk;
@@ -106,7 +106,7 @@ class ReducedVarianceModelBuilder : public Object {
 
     is_deprecated typename StatisticalModel<Representer>::Pointer BuildNewModelFromModel(const StatisticalModel<Representer>* model, double totalVariance) {
         statismo::StatisticalModel<Representer>* model_statismo = model->GetstatismoImplObj();
-        statismo::StatisticalModel<Representer>* new_model_statismo = callstatismoImpl(std::tr1::bind(&ImplType::BuildNewModelFromModel, this->m_impl, model_statismo, totalVariance));
+        statismo::StatisticalModel<Representer>* new_model_statismo = callstatismoImpl(boost::bind(&ImplType::BuildNewModelFromModel, this->m_impl, model_statismo, totalVariance));
         typename StatisticalModel<Representer>::Pointer model_itk = StatisticalModel<Representer>::New();
         model_itk->SetstatismoImplObj(new_model_statismo);
         return model_itk;

--- a/modules/ITK/include/itkStatisticalModel.h
+++ b/modules/ITK/include/itkStatisticalModel.h
@@ -48,7 +48,9 @@
 #include "StatisticalModel.h"
 #include "ModelInfo.h"
 #include "Representer.h"
-#include <boost/tr1/functional.hpp>
+
+#include <boost/bind.hpp>
+#include <boost/utility/result_of.hpp>
 
 namespace itk {
 
@@ -83,7 +85,7 @@ class StatisticalModel : public Object {
 
 
     template <class F>
-    typename std::tr1::result_of<F()>::type callstatismoImpl(F f) const {
+    typename boost::result_of<F()>::type callstatismoImpl(F f) const {
         try {
             return f();
         } catch (statismo::StatisticalModelException& s) {
@@ -143,142 +145,142 @@ class StatisticalModel : public Object {
     //TODO: wrap StatisticalModel* BuildReducedVarianceModel( double pcvar );
 
     const RepresenterType* GetRepresenter() const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::GetRepresenter, this->m_impl));
+        return callstatismoImpl(boost::bind(&ImplType::GetRepresenter, this->m_impl));
     }
 
     const DomainType& GetDomain() const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::GetDomain, this->m_impl));
+        return callstatismoImpl(boost::bind(&ImplType::GetDomain, this->m_impl));
     }
 
     DatasetPointerType DrawMean() const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::DrawMean, this->m_impl));
+        return callstatismoImpl(boost::bind(&ImplType::DrawMean, this->m_impl));
     }
 
     ValueType DrawMeanAtPoint(const PointType& pt) const {
         typedef ValueType (ImplType::*functype)(const PointType&) const;
-        return callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::DrawMeanAtPoint), this->m_impl, pt));
+        return callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::DrawMeanAtPoint), this->m_impl, pt));
     }
 
     ValueType DrawMeanAtPoint(unsigned ptid) const {
         typedef ValueType (ImplType::*functype)(unsigned) const;
-        return callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::DrawMeanAtPoint), this->m_impl, ptid));
+        return callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::DrawMeanAtPoint), this->m_impl, ptid));
     }
 
     DatasetPointerType DrawSample(const VectorType& coeffs, bool addNoise = false) const {
         typedef DatasetPointerType (ImplType::*functype)(const statismo::VectorType&, bool) const;
-        return callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::DrawSample), this->m_impl, fromVnlVector(coeffs), addNoise));
+        return callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::DrawSample), this->m_impl, fromVnlVector(coeffs), addNoise));
     }
 
     DatasetPointerType DrawSample(bool addNoise = false) const {
         typedef DatasetPointerType (ImplType::*functype)(bool) const;
-        return callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::DrawSample), this->m_impl, addNoise));
+        return callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::DrawSample), this->m_impl, addNoise));
     }
 
     DatasetPointerType DrawPCABasisSample(unsigned componentNumber) const {
         typedef DatasetPointerType (ImplType::*functype)(unsigned) const;
-        return callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::DrawPCABasisSample), this->m_impl, componentNumber));
+        return callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::DrawPCABasisSample), this->m_impl, componentNumber));
     }
 
     ValueType DrawSampleAtPoint(const VectorType& coeffs, const PointType& pt, bool addNoise = false) const {
         typedef ValueType (ImplType::*functype)(const statismo::VectorType&, const PointType&, bool) const;
-        return callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::DrawSampleAtPoint), this->m_impl, fromVnlVector(coeffs), pt, addNoise));
+        return callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::DrawSampleAtPoint), this->m_impl, fromVnlVector(coeffs), pt, addNoise));
     }
 
     ValueType DrawSampleAtPoint(const VectorType& coeffs, unsigned ptid, bool addNoise  = false) const  {
         typedef ValueType (ImplType::*functype)(const statismo::VectorType&, unsigned, bool) const;
-        return callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::DrawSampleAtPoint), this->m_impl, fromVnlVector(coeffs), ptid, addNoise));
+        return callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::DrawSampleAtPoint), this->m_impl, fromVnlVector(coeffs), ptid, addNoise));
     }
 
 
     VectorType ComputeCoefficientsForDataset(DatasetConstPointerType ds) const {
-        return toVnlVector(callstatismoImpl(std::tr1::bind(&ImplType::ComputeCoefficientsForDataset, this->m_impl, ds)));
+        return toVnlVector(callstatismoImpl(boost::bind(&ImplType::ComputeCoefficientsForDataset, this->m_impl, ds)));
     }
 
     VectorType ComputeCoefficientsForSample(DatasetConstPointerType ds) const {
-        return toVnlVector(callstatismoImpl(std::tr1::bind(&ImplType::ComputeCoefficientsForSample, this->m_impl, ds)));
+        return toVnlVector(callstatismoImpl(boost::bind(&ImplType::ComputeCoefficientsForSample, this->m_impl, ds)));
     }
 
     VectorType ComputeCoefficientsForDataSample(const DataItemType* sample) const {
-        return toVnlVector(callstatismoImpl(std::tr1::bind(&ImplType::ComputeCoefficientsForDataSample, this->m_impl, sample)));
+        return toVnlVector(callstatismoImpl(boost::bind(&ImplType::ComputeCoefficientsForDataSample, this->m_impl, sample)));
     }
 
     double ComputeLogProbabilityOfDataset(DatasetConstPointerType ds) const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::ComputeLogProbabilityOfDataset, this->m_impl, ds));
+        return callstatismoImpl(boost::bind(&ImplType::ComputeLogProbabilityOfDataset, this->m_impl, ds));
     }
 
     double ComputeProbabilityOfDataset(DatasetConstPointerType ds) const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::ComputeProbabilityOfDataset, this->m_impl, ds));
+        return callstatismoImpl(boost::bind(&ImplType::ComputeProbabilityOfDataset, this->m_impl, ds));
     }
 
     double ComputeLogProbabilityOfCoefficients(const VectorType& coeffs) const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::ComputeLogProbabilityOfCoefficients, this->m_impl, fromVnlVector(coeffs)));
+        return callstatismoImpl(boost::bind(&ImplType::ComputeLogProbabilityOfCoefficients, this->m_impl, fromVnlVector(coeffs)));
     }
 
     double ComputeProbabilityOfCoefficients(const VectorType& coeffs) const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::ComputeProbabilityOfCoefficients, this->m_impl, fromVnlVector(coeffs)));
+        return callstatismoImpl(boost::bind(&ImplType::ComputeProbabilityOfCoefficients, this->m_impl, fromVnlVector(coeffs)));
     }
 
     double ComputeMahalanobisDistanceForDataset(DatasetConstPointerType ds) const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::ComputeMahalanobisDistanceForDataset, this->m_impl, ds));
+        return callstatismoImpl(boost::bind(&ImplType::ComputeMahalanobisDistanceForDataset, this->m_impl, ds));
     }
 
     VectorType ComputeCoefficientsForPointValues(const PointValueListType& pvlist, double variance) const {
         typedef statismo::VectorType (ImplType::*functype)(const PointValueListType&, double) const;
-        return toVnlVector(callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::ComputeCoefficientsForPointValues), this->m_impl, pvlist, variance)));
+        return toVnlVector(callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::ComputeCoefficientsForPointValues), this->m_impl, pvlist, variance)));
     }
 
 
     DatasetPointerType DatasetToSample(DatasetConstPointerType ds) const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::DatasetToSample, this->m_impl, ds));
+        return callstatismoImpl(boost::bind(&ImplType::DatasetToSample, this->m_impl, ds));
     }
 
     unsigned GetNumberOfPrincipalComponents() const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::GetNumberOfPrincipalComponents, this->m_impl));
+        return callstatismoImpl(boost::bind(&ImplType::GetNumberOfPrincipalComponents, this->m_impl));
     }
 
     void Save(const char* modelname) {
         typedef void (ImplType::*functype)(const std::string&) const;
-        callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::Save), this->m_impl, modelname));
+        callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::Save), this->m_impl, modelname));
     }
 
     void Save(const H5::Group& modelRoot) {
         typedef void (ImplType::*functype)(const H5::Group&) const;
-        callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::Save), this->m_impl, modelRoot));
+        callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::Save), this->m_impl, modelRoot));
     }
 
 
     MatrixType GetCovarianceAtPoint(const PointType& pt1, const PointType& pt2) const {
         typedef statismo::MatrixType (ImplType::*functype)(const PointType&, const PointType&) const;
-        return  toVnlMatrix(callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::GetCovarianceAtPoint), this->m_impl, pt1, pt2)));
+        return  toVnlMatrix(callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::GetCovarianceAtPoint), this->m_impl, pt1, pt2)));
     }
 
     MatrixType GetCovarianceAtPoint(unsigned ptid1, unsigned  ptid2) const {
         typedef statismo::MatrixType (ImplType::*functype)(unsigned, unsigned ) const;
-        return toVnlMatrix(callstatismoImpl(std::tr1::bind(static_cast<functype>(&ImplType::GetCovarianceAtPoint),this->m_impl, ptid1, ptid2)));
+        return toVnlMatrix(callstatismoImpl(boost::bind(static_cast<functype>(&ImplType::GetCovarianceAtPoint),this->m_impl, ptid1, ptid2)));
     }
 
     MatrixType GetJacobian(const PointType& pt) const {
-        return toVnlMatrix(callstatismoImpl(std::tr1::bind(&ImplType::GetJacobian, this->m_impl, pt)));
+        return toVnlMatrix(callstatismoImpl(boost::bind(&ImplType::GetJacobian, this->m_impl, pt)));
     }
 
     MatrixType GetPCABasisMatrix() const {
-        return toVnlMatrix(callstatismoImpl(std::tr1::bind(&ImplType::GetPCABasisMatrix, this->m_impl)));
+        return toVnlMatrix(callstatismoImpl(boost::bind(&ImplType::GetPCABasisMatrix, this->m_impl)));
     }
 
     MatrixType GetOrthonormalPCABasisMatrix() const {
-        return toVnlMatrix(callstatismoImpl(std::tr1::bind(&ImplType::GetOrthonormalPCABasisMatrix, this->m_impl)));
+        return toVnlMatrix(callstatismoImpl(boost::bind(&ImplType::GetOrthonormalPCABasisMatrix, this->m_impl)));
     }
 
     VectorType GetPCAVarianceVector() const {
-        return toVnlVector(callstatismoImpl(std::tr1::bind(&ImplType::GetPCAVarianceVector, this->m_impl)));
+        return toVnlVector(callstatismoImpl(boost::bind(&ImplType::GetPCAVarianceVector, this->m_impl)));
     }
 
     VectorType GetMeanVector() const {
-        return toVnlVector(callstatismoImpl(std::tr1::bind(&ImplType::GetMeanVector, this->m_impl)));
+        return toVnlVector(callstatismoImpl(boost::bind(&ImplType::GetMeanVector, this->m_impl)));
     }
 
     const statismo::ModelInfo& GetModelInfo() const {
-        return callstatismoImpl(std::tr1::bind(&ImplType::GetModelInfo, this->m_impl));
+        return callstatismoImpl(boost::bind(&ImplType::GetModelInfo, this->m_impl));
     }
 
   private:

